File: ProAgil-App/src/app/_models/Evento.ts
import { Lote } from './Lote';
import { RedeSocial } from './RedeSocial';
import { Palestrante } from './Palestrante';

// Inferface de refentencia para trabalhar com um tipo especifico.
// export interface Evento {
// ↓ Tranformar em um classe para que o 'eventoEdit.component.ts possa receber um sua variável
// as atribuições.
export class Evento {

  constructor() { }

  id: number;
  local: string;
  dataEvento: Date;
  tema: string;
  qtdPessoas: number;
  imagemURL: string;
  telefone: string;
  email: string;
  lotes: Lote[];
  redesSociais: RedeSocial[];
  palestrantesEventos: Palestrante[];
}

File: ProAgil.Dominio/Evento.cs
using System;
using System.Collections.Generic;

namespace ProAgil.Dominio
{
    public class Evento
    {
        public int Id { get; set; }
        public string Local { get; set; }
        public DateTime DataEvento { get; private set; }
        public string Tema { get; set; }
        public int QtdPessoas { get; set; }
        public string ImagemURL { get; set; }
        public string Telefone { get; set; }
        public string Email { get; set; }

        // Quando colocamos uma lista quer dizer que a classe evento esta relacionada a outras classes (Lote, RedeSocial e PalestranteEvento)
        public List<Lote> Lotes { get; set; }
        public List<RedeSocial> RedeSociais { get; set; }
        public List<PalestranteEvento> PalestrantesEventos { get; set; }
    }
}